File: prog11.cpp
//container with most water
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

int maxArea(vector<int>& height)
{
    int n=height.size();
    int maxArea=0;
    int left=0;
    int right=n-1;

    while (left < right)
    {
        maxArea=fmax(maxArea, (right-left)*fmin(height[left],height[right]));
        if (height[left] < height[right])
            left++;
        else
            right--;
    }
    return maxArea;
}

int main()
{
    vector<int> height={1,2,3,4,5,6};
    cout << "the max area is " << maxArea(height) << endl;
    return 0;
}

File: prog11.py
def maxArea(height):
    n=len(height)
    maxArea=0
    left=0
    right=n-1

    while left<right:
        maxArea = max(maxArea, (right-left)*min(height[left],height[right]))
        if height[left] < height[right]:
            left+=1
        else:
            right-=1
    return maxArea


height=[1,2,3,4,5,6]
print (maxArea(height))
